### PR TITLE
UI classes improvements

### PR DESCRIPTION
**Pull request by 4z0t**
Removed overloaded methods doing same stuff as parent class and removed `OnFrame` of `Grid`, implemented lazyvar check and change. Test it to discover any bugs.

### CODE REVIEW

**Comment by 4z0t** [comment]
Don't merge it yet, `controls` folder has to be done too

**Comment by Garanas** [comment]
How about now? 😄

File: lua/ui/controls/acubutton.lua [hunk 1]
```diff
@@ -109,14 +109,15 @@ ACUButton = Class(Group) {
...
 
-        self.OnHide = function(self, hidden)
-            self.markerOverlay:SetHidden(hidden)
-            self.marker:SetHidden(hidden)
-            self.teamIndicator:SetHidden(hidden)
-            self.textOverlay:SetHidden(hidden)
-            return true
-        end
+    OnHide = function(self, hidden)
+        self:ApplyFunction(function (control)
```
Comment (Garanas) [inline lua/ui/controls/acubutton.lua:115]: This creates a closure.

Comment (4z0t) [reply lua/ui/controls/acubutton.lua:115]: This thing is in lobby only

**Comment by Garanas** [comment]
Is this one good enough for review?

**Comment by 4z0t** [comment]
> Is this one good enough for review?

@Garanas yeah, i think

**Comment by 4z0t** [comment]
@Ejsstiil ?

**Comment by Garanas** [comment]
I suspect he's reviewing it 😄

**Comment by radekzy** [comment]
Just resolving conflict with previously merged PR. Preserved your changes over mine.

**Comment by Ejsstiil** [comment]
Just fixed a conflict? Preserved your changes over mine. I guess that upon merging of this, no one would be sure which change is conforming both PR's

**Comment by Garanas** [comment]
this pull request should get merged though - I'm done checking half of it but the changes are difficult to confirm 😄

**Pull request merged**